### PR TITLE
Ripple effect touch

### PR DESCRIPTION
**Pull request by muhammedyalcin**
This branch contains ripple effect in touch util as an optional.

### CODE REVIEW

File: lib/touch.js [hunk 1]
```diff
@@ -78,8 +85,18 @@ function addPressEvent(target, event, options) {
         return handleTouch();
     };
 
+
+    let timeOut;
     target.onTouchEnded = (isInside) => {
-        endTouch(isInside);
+        timeOut !== undefined && clearTimeout(timeOut);
```
Comment (alperozisik) [inline lib/touch.js]:
```suggestion
        clearTimeout(timeOut);
```
No need to check: https://github.com/smartface/sf-extension-utils/blob/rippleEffectTouch/lib/base/timers.js#L40

File: doc/touch.md [hunk 2]
```diff
@@ -25,7 +25,12 @@ This replaces existing touch events
 **Params**: <code>object</code> [options] - Styling options  
 **Params**: <code>function</code> [options.startTouchEffect] - Function called when touch starts, to add UI effect to give pressed effects. If not provided, default effect will be used. It should be used with endTouchEffect  
 **Params**: <code>function</code> [options.endTouchEffect] - Function called when press effect ends; it is used to revert the effects in startTouchEffect. It should be used together with startTouchEffect. If not provided default effect reversing will be applied  
-**Params**: <code>boolean</code> [options.consumeTouch] - If this option is set to true, touch events won't be passed through views. If not provided, default value is undefined.  
+**Params**: <code>boolean</code> [options.consumeTouch] - If this option is set to true, touch events won't be passed through views. If not provided, default value is undefined.   
+**Params**: <code>boolean</code> [options.enableRippleEffect] - Enables the ripple effect on given target.
```
Comment (alperozisik) [inline doc/touch.md]: Do not modify this file manually. This file is to be autogenerated
```shell
npm run docs
```

Comment (alperozisik) [reply doc/touch.md]: https://github.com/smartface/sf-extension-utils/blob/cd34fcf5a872db85d2798ded60902400923ac72c/package.json#L7

File: lib/touch.js [hunk 3]
```diff
@@ -169,3 +186,32 @@ function defaultClearPressEffect() {
         target.backgroundColor = target.__pressEffectOriginalColor__;
     }
 }
+
```
Comment (alperozisik) [inline lib/touch.js]: 1) Do not add document this or mark it as internal. Externally this method cannot be used
2) The documentation and the function does not match
3) Add options to the `applyRippleEffect` part of the documentation

File: doc/touch.md [hunk 4]
```diff
@@ -30,6 +30,9 @@ This replaces existing touch events
 **Params**: <code>number</code> [options.rippleDuration] - Ripple effect requires duration before performing given event. This option specfic to Android.  
 **Params**: <code>boolean</code> [options.rippleUseForeground] - iff this options is set to true, ripple effect added on top of the given target. No matter what given target contains, draw ripple effect over them. This option specfic to Android. By default, ripple effect is given.  
 **Params**: <code>UI.Color</code> [options.rippleColor] - Sets the color to ripple effect. This option specfic to Android.  
+**Params**: <code>UI.Color</code> [options.fadeColor] - Sets the color to fade effect. This option specfic to iOS.
```
Comment (doganekici) [inline doc/touch.md]: Requires iOS emulator & player 6.10.1-beta.33

**Pull request merged**